api/v1/submissions/PKPSubmissionFileHandler.inc.php

```diff
@@ -196,23 +197,25 @@ public function getMany($slimRequest, $response, $args)
             }
         }
 
-        \HookRegistry::call('API::submissions::files::params', [&$params, $slimRequest]);
-
-        $params['submissionIds'] = [$this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION)->getId()];
-
+        $collector = Repo::submissionFiles()
+            ->getCollector()
+            ->filterBySubmissionIds(
+                [$submission->getId()]
+            )
+            ->filterByReviewRoundIds($allowedReviewRoundIds)
```

Comment:
Or at [Collector, line 201 ](https://github.com/pkp/pkp-lib/pull/7166/files#diff-1fbcec3f507f0b4e1ea4afe98da72aec2b9a57511c64cee696767b31f5e98f48R210), we could implement another filtering for `$this->reviewRoundIds !== []`.

There's any scenario where is valid to do a `IN` with empty values? I don't think so.

Reply:
> There's any scenario where is valid to do a IN with empty values? I don't think so.

No, there's not, and I believe Laravel will throw an exception if it tries to execute a `whereIn` with an empty array. But I'd prefer to keep the `Collector::filterBy...()` syntaxes consistent across the board. `null` = do not apply this filter, `array` = apply this filter. Let the collector throw an exception and the code that's calling it can be fixed at source, instead of trying to build an exception into the `Collector`.

Reply:
Okay!